core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddress.java

```diff
@@ -77,7 +77,7 @@ public void doWriteObject(ObjectOutput output, JGroupsAddress address) throws IO
       public JGroupsAddress doReadObject(ObjectInput unmarshaller) throws IOException, ClassNotFoundException {
          try {
             org.jgroups.Address address = org.jgroups.util.Util.readAddress(unmarshaller);
-            return (JGroupsAddress) JGroupsAddressCache.fromJGroupsAddress(address);
+            return (JGroupsAddress) JGroupsAddressCache.fromJGroupsAddressUnchecked(address);
```

Comment:
exactly. If you only instantiate this class, the old code works fine and this change isn't needed :)